README.md

```diff
@@ -180,16 +180,55 @@ npm test
 
 ### Important notes
 
-- Next.js now uses Rust-based compiler [SWC to compile](https://nextjs.org/docs/upgrading#swc-replacing-babel) JavaScript/TypeScript. This new compiler is up to 17x faster than Babel when compiling individual files and up to 5x faster Fast Refresh. Next.js provides full backward compatibility with applications that have custom Babel configuration.
-- Next.js 12 HMR connection now uses a [WebSocket](https://nextjs.org/docs/upgrading#nextjs-hmr-connection-now-uses-a-websocket).
+- Next.js now uses Rust-based compiler [SWC to compile][swc-replacing-babel] JavaScript/TypeScript. This new compiler is up to 17x faster than Babel when compiling individual files and up to 5x faster Fast Refresh. Next.js provides full backward compatibility with applications that have custom Babel configuration.
+- Next.js 12 HMR connection now uses a [WebSocket][websocket].
 
-### Breaking Changes
+#### Middleware
+
+This feature permit you run code before a request is completed. Based on the user's incoming request, you can modify the response by rewriting, redirecting, adding headers, or even streaming HTML.
+
+It works out of the box on Edge platforms like Vercel, which uses Edge Functions. However, this feature will work differently on Vercel and VIP because it will hit a different cache between the edges. Since it’s still an experimental feature, we're not recommending the usage until the feature will be more useful.
+
+#### URL Imports
+
+You can now import tooling directly from the CDN without different builds or installs. In addition, we will be able to import those from the wire rather than build them locally.
+
+It’s good to import things like static Images, but it is preferable if the organization has a proper CDN or control over the content of the files.
+
+If you want to proceed with the feature, don't forget to add the URL prefix in the `next.config.js` file:
+
+```js
+module.exports = {
+  experimental: {
+    urlImports: ['https://www.wpvip.com/']
+  }
+}
+```
 
-- Webpack 4 support has been removed. See the [webpack 5 upgrading documentation](https://nextjs.org/docs/messages/webpack5) for more info.
-- Next target option has been deprecated. If you are currently using the target option set to serverless please read the [documentation on how to leverage the new output](https://nextjs.org/docs/advanced-features/output-file-tracing).
-- Next/image changed wrapping element. See the [documentation](https://nextjs.org/docs/basic-features/image-optimization#styling) for more info.
-- The minimum [Node.js](https://nodejs.org/en/) version has been bumped from 12.0.0 to 12.22.0 which is the first version of Node.js with native ES Modules support.
+To prevent errors from TypeScript, you also need to create a `types/skypack.d.ts` file in the root of the project to permit import from external URLs, as the example below:
```

Comment:
```suggestion
To prevent TypeScript errors, you also need to create a `types/skypack.d.ts` file at the root of the project. The following is an example of that:
```